reclass/datatypes/parameters.py

```diff
@@ -134,19 +153,42 @@ def _merge_dict(self, cur, new, path):
             ret.update(new)
             return ret
 
+        ovrprfx = Parameters.DICT_KEY_OVERRIDE_PREFIX
+
         for key, newvalue in new.iteritems():
-            ret[key] = self._merge_recurse(ret.get(key), newvalue,
-                                           path.new_subpath(key))
+            if key.startswith(ovrprfx) and not initmerge:
+                ret[key.lstrip(ovrprfx)] = newvalue
+            else:
```

Comment:
Hmm... I think I see what you're saying here. This certainly does present a practical implementation challenge. I think I need to get my hands dirty a bit with this feature to see how it all plays out.